### DISCUSSION TITLE
Import "numpy" could not be resolved.

### DISCUSSION THREAD

**Discussion by ianyepan** [microsoft/pyright#1721] [Q&A]
I'm using pyright as a LSP server in Emacs (via the lsp-mode package), and pyright is having trouble resolving pip3 installed packages like numpy or matplotlib. I checked numpy's installation location and it's under `/home/ianpan/.local/lib/python3.8/site-packages`, and when I ran `pyright --verbose` I do see this exact same path being found both under "Search paths found for configured python interpreter" and "Received 5 paths from interpreter".
In this case, is the `pyrightconfig.json` file still needed? If it is, what exactly should I put in the configuration file? All I need is for pyright to recognize my pip3 packages that installed "globally". Thanks in advance!

**Comment by erictraut** [comment]
You shouldn't need to specify your python environment in `pyrightconfig.json`. Python will use the default python environment configured in your shell. It does this by executing "python3" and running a small script in the selected Python interpreter to dump the paths the `sys.path`. From your description, it sounds like it is finding these properly. Those are the paths it lists under "Received 5 paths from interpreter".

When you say "having trouble resolving installed packages", what symptoms are you seeing?

With the `--verbose` flag enabled, pyright should output details when it attempts and fails to resolve a module. Are there any other clues in the verbose output?

**Reply by ianyepan** [reply]
## Solved
Thanks for the fast reply @erictraut ! The symptoms I was seeing was an import resolution error, i.e. "**Import numpy could not be resolved**" etc. You mentioned that pyright **executes "python3" and runs a small script in the selected Python interpreter to dump the paths the sys.path**, and it turns out the command "python3" was exactly the problem! Emacs' LSP client for pyright "lsp-pyright" via "lsp-mode" defaults the command to "python", which points to python2 on my system. So after adding the following configuration in my Emacs' setting file `init.el`, the problem went away:

```emacs-lisp
(setq lsp-pyright-python-executable-cmd "python3")
```

For fellow Emacs users seeking the answer, the following snippet is the complete solution to have Pyright working in Emacs:

```emacs-lisp
(use-package python
  :config
  (setq python-shell-interpreter "python3"))

(use-package lsp-pyright
  :hook (python-mode . (lambda () (require 'lsp-pyright)))
  :init (when (executable-find "python3")
          (setq lsp-pyright-python-executable-cmd "python3")))
```
